CMakeLists.txt

```diff
@@ -37,6 +58,24 @@ if(MSVC)
   # for recent C++ language standards. Without this option MSVC returns the
   # value of __cplusplus="199711L"
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
+
+  # When using vcpkg, all targets build with the same runtime
+  if(VCPKG_TOOLCHAIN)
+    if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+      set(VCPKG_TARGET_TRIPLET
+          "$ENV{VCPKG_DEFAULT_TRIPLET}"
+          CACHE STRING "")
+    endif()
+    if(VCPKG_TARGET_TRIPLET MATCHES "^.*windows-static$")
+      set(CMAKE_MSVC_RUNTIME_LIBRARY
+          "MultiThreaded$<$<CONFIG:Debug>:Debug>"
```

Comment:
Why is it only Debug?

Reply:
These codes mean set ```CMAKE_MSVC_RUNTIME_LIBRARY``` to ```MultiThreadedDebug``` when ```CMAKE_BUILD_TYPE``` is ```Debug``` , otherwise set ```CMAKE_MSVC_RUNTIME_LIBRARY``` to ```MultiThreaded``` . Just like https://github.com/microsoft/vcpkg/blob/master/scripts/toolchains/windows.cmake#L3 in [vcpkg](https://github.com/microsoft/vcpkg/blob/master/scripts/toolchains/windows.cmake#L3).

These codes are a old patch to compact for both ```x64/x86-windows-static``` and ```x64/x86-windows-static-md``` and can be replaced by the new codes ```set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<STREQUAL:${VCPKG_CRT_LINKAGE},dynamic>:DLL>" CACHE STRING "")``` now. I will update them later.

Reply:
@maxgolov I have already update the cmake script and rebased from master again.